transaction.go

```diff
@@ -222,6 +232,11 @@ func (t *Transaction) payloadCanonicalForm() interface{} {
 		authorizers[i] = auth.Bytes()
 	}
 
+	arguments := make([][]byte, len(t.Arguments))
+	for i, arg := range t.Arguments {
+		arguments[i], _ = jsoncdc.Encode(arg)
```

Comment:
Is the second return value here an error? If so should maybe be returned?

Or is it assumed the arguments are all valid here? Maybe then assert/panic if they aren't?

Don't quite know what this `payloadCanonicalForm` function is used for

Reply:
Yeah, it's an error. 

> Or is it assumed the arguments are all valid here?

Unless there is a bug in our code or somebody manually constructs an invalid `cadence.Value` object, it shouldn't be possible to pass an invalid value. I struggle a bit with cases like this: do we inconvenience the happy path (ID generation, hashing, signatures) in order to catch a case that only occurs under misuse? 

For now I'd like to add a panic so that it doesn't fail silently, but I also want to give some more thought to how we handle and propagate these kinds of errors.

Reply:
Right, I think the errors produced from the encoding, like composites with mismatching field values and field types, are "just" going to cause invalid transactions, which should be fine. 

I guess the error can just be bubbled up, but like you wrote, maybe should be thought through more and done in a separate PR 👍